cmd/tsconnect/wasm/wasm_js.go

```diff
@@ -69,6 +69,12 @@ func newIPN(jsConfig js.Value) map[string]any {
 		store = &jsStateStore{jsStateStorage}
 	}
 
+	jsAuthKey := jsConfig.Get("authKey")
+	var authKey string
+	if !jsAuthKey.IsUndefined() && jsAuthKey.Type() == js.TypeString {
```

Comment:
I think `jsAuthKey.Type() == js.TypeString` is all you need. Undefined is its own type in JS (and mapped to `js.TypeUndefined` in the Go JS bindings).